AK/GenericLexer.cpp

```diff
@@ -128,6 +129,62 @@ StringView GenericLexer::consume_quoted_string(char escape_char)
     return m_input.substring_view(start, length);
 }
 
+template<Integral T>
+ErrorOr<T, IntegerParseError> GenericLexer::consume_decimal_integer()
+{
+    using UnsignedT = MakeUnsigned<T>;
+
+    ArmedScopeGuard rollback { [&, rollback_position = m_index] {
```

Comment:
This feels like a future PR could extend `AK::TemporaryChange` to do this, or create a distinct type to do this. Pity we can't do the ol SCOPE_FAIL trick as we don't use exceptions :)

Reply:
Maybe this could be like... 
```
#define ROLLBACK(error) ({
    m_index = rollback_position;
    error;
})
```

or something? ... meh. Bikeshedding is more fun than approving changes